go-controller/pkg/ovn/controller/apbroute/node_controller.go

```diff
@@ -48,27 +47,27 @@ type ExternalGatewayNodeController struct {
 	namespaceLister   corev1listers.NamespaceLister
 	namespaceInformer cache.SharedIndexInformer
 
-	//external gateway caches
-	//make them public so that they can be used by the annotation logic to lock on namespaces and share the same external route information
-	ExternalGWCache map[ktypes.NamespacedName]*RouteInfo
-	ExGWCacheMutex  *sync.RWMutex
+	routePolicyFactory adminpolicybasedrouteinformer.SharedInformerFactory
 
 	mgr *externalPolicyManager
 }
 
 func NewExternalNodeController(
+	apbRoutePolicyClient adminpolicybasedrouteclient.Interface,
 	podInformer coreinformers.PodInformer,
 	namespaceInformer coreinformers.NamespaceInformer,
-	apbRouteInformer adminpolicybasedrouteinformer.AdminPolicyBasedExternalRouteInformer,
 	stopCh <-chan struct{},
 ) (*ExternalGatewayNodeController, error) {
 
 	namespaceLister := namespaceInformer.Lister()
+	routePolicyFactory := adminpolicybasedrouteinformer.NewSharedInformerFactory(apbRoutePolicyClient, resyncInterval)
+	externalRouteInformer := routePolicyFactory.K8s().V1().AdminPolicyBasedExternalRoutes()
 
```

Comment:
I am not sure if this change was intended, or just a result of some rebase, but most of the changes to "unused variables in node_controller.go" probably should be reverted. We use shared Factory now, and I guess the e2e failure is caused by us not starting the `routePolicyFactory` you created here, so node_controller is not handling any events

Reply:
I'm not sure about that since it is initialized with an apb client, so it should get the events. What it is probably doing wrong is duplicating the informer and the memory consumption by this.
I will change this to use the shared informer as you suggested. I don't remember why this was changed during the cleanup.